### PR TITLE
[JENKINS-72112] Allow merge strategy inheritance in PodTemplates

### PR DESCRIPTION
**Pull request by georgblumenschein**
Ticket: https://issues.jenkins.io/browse/JENKINS-72112

### Note

**This Pull Request is taken from https://github.com/jenkinsci/kubernetes-plugin/pull/1448 and the changes done there!**
I was not quite sure what the workflow in such cases is, I could also open a PR on the fork so that the first forker can update the PR here, I just assumed it is simpler this way.

### Testing done

Testing done in https://github.com/jenkinsci/kubernetes-plugin/pull/1448, I am fixing (at least I try ;-) ) the automated tests that prevent it from merging.

```[tasklist]
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by georgblumenschein** [comment]
Two notes:
I had to rebase to get the pipeline green, meaning it looks like I did that, but credit for 99% of the work goes to @vivienfricadelamadeus 
Failing test seems to be flaky with a problem pulling the busybox image, all checks ran through before I squashed them, so a retrigger should resolve this.

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html [hunk 1]
```diff
@@ -0,0 +1 @@
+False by default
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/help-inheritYamlMergeStrategy.html:1]:
```suggestion
Controls whether the defined yaml merge strategy will be inherited if another defined pod template is configured to inherit from the current one.
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly [hunk 2]
```diff
@@ -87,7 +87,11 @@ THE SOFTWARE.
     <f:textarea/>
   </f:entry>
 
-  <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
+    <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
+
+    <f:entry field="inheritYamlMergeStrategy" title="${%Inherit yaml merge strategy}" >
+      <f:checkbox default="false"/>
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly:93]:
```suggestion
      <f:checkbox/>
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly [hunk 3]
```diff
@@ -87,7 +87,11 @@ THE SOFTWARE.
     <f:textarea/>
   </f:entry>
 
-  <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
+    <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly:90]:
```suggestion
  <f:dropdownDescriptorSelector title="${%Yaml merge strategy}" field="yamlMergeStrategy" default="${descriptor.defaultYamlMergeStrategy}" />
```

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**